src/utils/fns.js

```diff
@@ -2,4 +2,16 @@ function random(limit) {
   return Math.floor(Math.random() * limit);
 }
 
-export { random };
+function formatTitleWithApostrophe(title) {
+  let result = title;
+  while (title.includes('&#39;')) {
```

Comment:
Here's the bug that's causing an infinite loop in your code, you should change this line:

```suggestion
  while (result.includes('&#39;')) {
```